.github/workflows/test-gutenberg-build-process-run.yml

```diff
@@ -0,0 +1,100 @@
+##
+# A callable workflow that tests the Gutenberg plugin build process.
+##
+name: Test the Gutenberg plugin Build Process
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: 'Operating system to run tests on'
+        required: false
+        type: 'string'
+        default: 'ubuntu-latest'
+      with-core:
+        description: 'Whether to test with WordPress Core'
+        required: false
+        type: 'boolean'
+        default: false
+      directory:
+        description: 'Directory to run WordPress out of. Valid values are src or build'
+        required: false
+        type: 'string'
+        default: 'src'
+
+env:
+  GUTENBERG_DIRECTORY: ${{ inputs.directory == 'build' && 'build' || 'src' }}/wp-content/plugins/gutenberg
+  PUPPETEER_SKIP_DOWNLOAD: ${{ true }}
+
+jobs:
+  # Verifies that installing npm dependencies and building the Gutenberg plugin works as expected.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Checks out the Gutenberg plugin repository in the plugins directory.
+  # - Sets up Node.js.
+  # - Logs debug information about the GitHub Action runner.
+  # - Installs Core npm dependencies if testing with Core.
+  # - Installs npm dependencies for Gutenberg.
+  # - Runs the Gutenberg build process.
+  # - Builds WordPress to run from the relevant location (src or build) if testing with Core.
+  # - Runs the Gutenberg build process.
+  # - Ensures version-controlled files are not modified or deleted.
+  build-process-tests:
+    name: Gutenberg ${{ inputs.with-core && ' with Core' || '' }} / Running from ${{ inputs.directory }} on ${{ inputs.os == 'macos-latest' && 'MacOS' || inputs.os == 'windows-latest' && 'Windows' || 'Linux' }}
+    runs-on: ${{ inputs.os }}
+    timeout-minutes: 30
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+
+      - name: Checkout Gutenberg plugin
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+        with:
+          repository: 'WordPress/gutenberg'
+          path: ${{ env.GUTENBERG_DIRECTORY }}
+          show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@5e21ff4d9bc1a8cf6de233a3057d20ec6b3fb69d # v3.8.1
+        with:
+          node-version-file: '.nvmrc'
```

Comment:
What if core and Gutenberg don't use the same Node.js version? I know we try to keep them in sync, but it could be possible that they aren't.

Reply:
I guess that's possible. Even if for a few commits. We could run `nvm install` in between to make sure the right version is active.